### PR TITLE
Fix size_t inconsistencies

### LINKED ISSUE #460: Crash while building
While building on my computer I get the following error:

```
[ 22%] Building CXX object src/manifold/CMakeFiles/manifold.dir/src/boolean_result.cpp.o
cd /app/manifold/build/src/manifold && /usr/bin/c++ -Dmanifold_EXPORTS -I/app/manifold/src/manifold/include -I/app/manifold/src/utilities/include -I/app/manifold/src/third_party/thrust/dependencies/libcudacxx/include -I/app/manifold/src/third_party/thrust -I/app/manifold/src/third_party/glm/glm/.. -I/app/manifold/src/cross_section/include -I/app/manifold/src/third_party/clipper2/CPP/Clipper2Lib/include -I/app/manifold/src/collider/include -I/app/manifold/src/polygon/include -I/app/manifold/src/third_party/graphlite/include -O3 -DNDEBUG -fPIC -fPIC -Werror -Wall -Wno-sign-compare -Wno-unused -O3 -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_CPP -std=gnu++17 -MD -MT src/manifold/CMakeFiles/manifold.dir/src/boolean_result.cpp.o -MF CMakeFiles/manifold.dir/src/boolean_result.cpp.o.d -o CMakeFiles/manifold.dir/src/boolean_result.cpp.o -c /app/manifold/src/manifold/src/boolean_result.cpp
In file included from /app/manifold/src/utilities/include/sparse.h:22,
                 from /app/manifold/src/collider/include/collider.h:17,
                 from /app/manifold/src/manifold/src/impl.h:18,
                 from /app/manifold/src/manifold/src/boolean3.h:16,
                 from /app/manifold/src/manifold/src/boolean_result.cpp:18:
In static member function ‘static void manifold::ManagedVec<T>::mallocManaged(T**, size_t) [with T = int]’,
    inlined from ‘std::tuple<manifold::VecDH<int>, manifold::VecDH<int> > {anonymous}::SizeOutput(manifold::Manifold::Impl&, const manifold::Manifold::Impl&, const manifold::Manifold::Impl&, const manifold::VecDH<int>&, const manifold::VecDH<int>&, const manifold::VecDH<int>&, const manifold::VecDH<int>&, const manifold::SparseIndices&, const manifold::SparseIndices&, bool, manifold::ExecutionPolicy)’ at /app/manifold/src/utilities/include/vec_dh.h:155:20:
/app/manifold/src/utilities/include/vec_dh.h:261:42: error: argument 1 value ‘18446744073709551612’ exceeds maximum object size 9223372036854775807 [-Werror=alloc-size-larger-than=]
       *ptr = reinterpret_cast<T *>(malloc(bytes));
                                    ~~~~~~^~~~~~~
In file included from /usr/include/c++/8/cstdlib:75,
                 from /usr/include/c++/8/bits/stl_algo.h:59,
                 from /usr/include/c++/8/algorithm:62,
                 from /app/manifold/src/manifold/src/boolean_result.cpp:15:
/usr/include/stdlib.h: In function ‘std::tuple<manifold::VecDH<int>, manifold::VecDH<int> > {anonymous}::SizeOutput(manifold::Manifold::Impl&, const manifold::Manifold::Impl&, const manifold::Manifold::Impl&, const manifold::VecDH<int>&, const manifold::VecDH<int>&, const manifold::VecDH<int>&, const manifold::VecDH<int>&, const manifold::SparseIndices&, const manifold::SparseIndices&, bool, manifold::ExecutionPolicy)’:
/usr/include/stdlib.h:539:14: note: in a call to allocation function ‘void* malloc(size_t)’ declared here
 extern void *malloc (size_t __size) __THROW __attribute_malloc__ __wur;
              ^~~~~~
cc1plus: all warnings being treated as errors
src/manifold/CMakeFiles/manifold.dir/build.make:92: recipe for target 'src/manifold/CMakeFiles/manifold.dir/src/boolean_result.cpp.o' failed
make[2]: *** [src/manifold/CMakeFiles/manifold.dir/src/boolean_result.cpp.o] Error 1
make[2]: Leaving directory '/app/manifold/build'
CMakeFiles/Makefile2:668: recipe for target 'src/manifold/CMakeFiles/manifold.dir/all' failed
make[1]: *** [src/manifold/CMakeFiles/manifold.dir/all] Error 2
make[1]: Leaving directory '/app/manifold/build'
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

Any idea what might cause this?

### PR DESCRIPTION
**Pull request by pca006132**
For some reason `vec_dh.h` was mixing size_t and int for indices and vector size. This is now fixed to use `size_t` everywhere, which should allow large meshes in the future (note that most of the functions/data structure still expect 32 bit integer indices so large meshes are not yet supported). As I cannot see `countAt` that uses long, so I hardcoded it to use 32 bit integers (otherwise we need a lot of long to int casts).

This also disabled the alloc-size-larger-than warning which causes #450 and #460. I originally thought that the warning might be related to mixing 32 bit and 64 bit signed/unsigned integers but it seems not related.

Fixes #460

@emil-peters @geoffder can you help me test if this can compile on your machine?

### CODE REVIEW

**Comment by geoffder** [comment]
Builds and passes tests on my machine :+1:

File: src/utilities/src/util.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <thrust/iterator/counting_iterator.h>
+
+namespace manifold {
+thrust::counting_iterator<int> countAt(int i) {
+  return thrust::make_counting_iterator(i);
```
Comment (elalish) [inline src/utilities/src/util.cpp:18]: Do we really need a whole extra file for this tiny function? Can't we just inline it in the header?

**Review by elalish — COMMENTED** [review]
Thanks for the update!

**Comment by elalish** [comment]
If you put "Fixes #460" in the description, it'll be closed automatically when this PR is merged.

File: src/utilities/src/util.cpp (hunk 1 above)
Comment (pca006132) [reply src/utilities/src/util.cpp:18]: True, I will change it into an inline function

**Comment by emil-peters** [comment]
I don't think you're gonna like to hear this, but I'm now getting a similar error at a different file :sweat_smile: 
```
make[2]: Entering directory '/app/manifold/build'
[ 25%] Building CXX object src/cross_section/CMakeFiles/cross_section.dir/src/cross_section.cpp.o
cd /app/manifold/build/src/cross_section && /usr/bin/c++  -I/app/manifold/src/cross_section/include -I/app/manifold/src/utilities/include -I/app/manifold/src/third_party/thrust -I/app/manifold/src/third_party/thrust/dependencies/libcudacxx/include -I/app/manifold/src/third_party/glm/glm/.. -I/app/manifold/src/third_party/clipper2/CPP/Clipper2Lib/include -O3 -DNDEBUG -fPIC -Werror -Wall -Wno-sign-compare -Wno-unused -Wno-alloc-size-larger-than -O3 -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_CPP -std=gnu++17 -MD -MT src/cross_section/CMakeFiles/cross_section.dir/src/cross_section.cpp.o -MF CMakeFiles/cross_section.dir/src/cross_section.cpp.o.d -o CMakeFiles/cross_section.dir/src/cross_section.cpp.o -c /app/manifold/src/cross_section/src/cross_section.cpp
/app/manifold/src/cross_section/src/cross_section.cpp: In member function ‘manifold::CrossSection manifold::CrossSection::RectClip(const manifold::Rect&) const’:
/app/manifold/src/cross_section/src/cross_section.cpp:402:17: error: ‘ExecuteRectClip’ is not a member of ‘C2’
   auto ps = C2::ExecuteRectClip(r, GetPaths(), false, precision_);
                 ^~~~~~~~~~~~~~~
/app/manifold/src/cross_section/src/cross_section.cpp:402:17: note: suggested alternative: ‘RectClip’
   auto ps = C2::ExecuteRectClip(r, GetPaths(), false, precision_);
                 ^~~~~~~~~~~~~~~
                 RectClip
At global scope:
cc1plus: error: unrecognized command line option ‘-Wno-alloc-size-larger-than’ [-Werror]
cc1plus: all warnings being treated as errors
src/cross_section/CMakeFiles/cross_section.dir/build.make:78: recipe for target 'src/cross_section/CMakeFiles/cross_section.dir/src/cross_section.cpp.o' failed
make[2]: *** [src/cross_section/CMakeFiles/cross_section.dir/src/cross_section.cpp.o] Error 1
make[2]: Leaving directory '/app/manifold/build'
CMakeFiles/Makefile2:639: recipe for target 'src/cross_section/CMakeFiles/cross_section.dir/all' failed
make[1]: *** [src/cross_section/CMakeFiles/cross_section.dir/all] Error 2
make[1]: Leaving directory '/app/manifold/build'
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by pca006132** [comment]
This is weird... They check for alloc size but does not allow disabling the check...
Maybe we should just recommend GCC9 or later.

For the compilation error related to cross_section however, it should be a problem with submodule not being up-to-date (or not being outdated because I forgot to merge the latest master branch)

**Comment by pca006132** [comment]
Closing now as I need to think about how to fix this...

**Pull request closed without merging**